src/pages/index.js

```diff
@@ -0,0 +1,330 @@
+    handleImageClick(data);
+  });
+
+  return cardElement;
+}
+
+function handleCardLike(evt, dataId) {
+  evt.target.classList.toggle("card__like-button_liked");
+
+  let isLiked = evt.target.classList.contains("card__like-button_liked")
+    ? true
+    : false;
+  api.toggleLike(dataId, isLiked).then().catch(console.error);
+}
+
+function handleImageClick(data) {
+  openModal(previewModal);
+  previewModalImageEl.src = data.link;
+  previewModalCaptionEl.textContent = data.name;
+  previewModalImageEl.alt = data.name;
+}
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+
+  deleteForm.addEventListener("submit", handleDeleteSubmit);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  submitButton.textContent = "Deleting...";
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitButton.textContent = "Delete";
+      closeModal(deleteModal);
+    });
+}
+
+function handleOutsideClick(modal) {
+  return function (event) {
+    if (!event.target.closest(".modal__content")) {
+      closeModal(modal);
+    }
+  };
+}
+
+function handleEscapePress(modal) {
+  return function (event) {
+    if (event.key === "Escape" || event.key === "Esc") {
+      closeModal(modal);
+    }
+  };
+}
+
+let outsideClickHandler, escapePressHandler;
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+
+  outsideClickHandler = handleOutsideClick(modal);
+  escapePressHandler = handleEscapePress(modal);
+
+  setTimeout(() => {
+    addListeners(outsideClickHandler, escapePressHandler);
+  }, 1);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+
+  if (outsideClickHandler && escapePressHandler) {
+    removeListeners(outsideClickHandler, escapePressHandler);
+  }
+
+  outsideClickHandler = null;
+  escapePressHandler = null;
+}
+
+function addListeners(outsideClickHandler, escapePressHandler) {
+  document.addEventListener("click", outsideClickHandler);
+  document.addEventListener("keydown", escapePressHandler);
+}
+
+function removeListeners(outsideClickHandler, escapePressHandler) {
+  document.removeEventListener("click", outsideClickHandler);
+  document.removeEventListener("keydown", escapePressHandler);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  submitButton.textContent = "Saving...";
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitButton.textContent = "Save";
+      disableButton(profileSubmitButton, settings);
```

Comment:
`      disableButton(profileSubmitButton, settings);
`

You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.

Reply:
**Please, correct it everywhere in the code**

Reply:
Fixed this too. I just did the change in text for the finally for each case.